lib/rules/selector-max-compound-selectors/index.mjs

```diff
@@ -64,34 +65,35 @@ const rule = (primary, secondaryOptions) => {
 		 * @param {import('postcss').Rule} ruleNode
 		 */
 		function checkSelector(selectorNode, ruleNode) {
-			let compoundCount = 1;
+			/** @type {import('postcss-selector-parser').Node[]} */
+			const filteredChildNodes = [];
 
-			selectorNode.each((childNode, index) => {
+			selectorNode.each((childNode) => {
 				// Only traverse inside actual selectors and context functional pseudo-classes
-				if (childNode.type === 'selector' || isContextFunctionalPseudoClass(childNode)) {
+				if (isSelector(childNode) || isContextFunctionalPseudoClass(childNode)) {
 					checkSelector(childNode, ruleNode);
 				}
 
-				// Compound selectors are separated by combinators, so increase count when meeting one
-				if (childNode.type === 'combinator') {
-					compoundCount++;
-
-					return;
-				}
-
-				// Try ignoring the selector if the current node is the first, or the previous node is a combinator
-				const previousNode = selectorNode.at(index - 1);
-
-				if ((!previousNode || previousNode.type === 'combinator') && isSelectorIgnored(childNode)) {
-					compoundCount--;
+				if (!isSelectorIgnored(childNode)) {
+					filteredChildNodes.push(childNode);
 				}
 			});
 
-			if (
-				selectorNode.type !== 'root' &&
-				selectorNode.type !== 'pseudo' &&
-				compoundCount > primary
-			) {
+			// Normalize selector nodes and count combinators
+			const combinatorCount = filteredChildNodes
+				.filter((node, i, nodes) => {
+					// Filter out a consecutive combinator
+					return !(isCombinator(node) && i > 0 && isCombinator(nodes[i - 1]));
```

Comment:
minor

```js
const isLeftover = isCombinator(node) && i > 0 && isCombinator(nodes[i - 1]);

return !isLeftover;
```

<hr />

refactor

It doesn't matter which one you remove, but I would have expected to remove the left one.
That would require to check whether `i + 1` exists which would enable to have only 2 filters.
i.e. the l91 one could be performed at the same time
e.g. `return !(isLeftover || isFirst || isLast);`

Reply:
@Mouvedia Sorry for the late reply. Based on your feedback, I just tried refactoring (b7a723440).